adhocracy4/polls/assets/PollDashboard/EditPollManagement.jsx

```diff
@@ -104,33 +104,39 @@ export const EditPollManagement = (props) => {
     }
   }
 
-  const handleChoice = (action, params) => {
+  const handleChoiceLabel = (index, choiceIndex, label) => {
     const diff = {}
-    if (action === 'label') {
-      const { index, choiceIndex, label } = params
-      diff[index] = { choices: {} }
-      diff[index].choices[choiceIndex] = { $merge: { label } }
-    } else if (action === 'append') {
-      const { index, hasOtherOption } = params
-      const position = questions[index].choices.length - 1
-      const newChoice = getNewChoice()
-      diff[index] = hasOtherOption
-        ? { choices: { $splice: [[position, 0, newChoice]] } }
-        : { choices: { $push: [newChoice] } }
-    } else if (action === 'is-other-choice') {
-      const { index, isOtherChoice } = params
-      if (isOtherChoice) {
-        const otherChoice = getNewChoice('other', true)
-        diff[index] = { choices: { $push: [otherChoice] } }
-      } else {
-        const choiceIndex = questions[index].choices.findIndex(c => c.key === 'other-choice')
-        diff[index] = { choices: { $splice: [[choiceIndex, 1]] } }
-      }
-    } else if (action === 'delete') {
-      const { index, choiceIndex } = params
+    diff[index] = { choices: {} }
+    diff[index].choices[choiceIndex] = { $merge: { label } }
```

Comment:
more of a general thing: i find these `immutability-helper` functions quite hard to read and to understand. And we do not use them consistently. I think we should either get rid of them entirely, or use them more consistently. What do you think?

Reply:
I also don't really like them and I do agree with standardising the usage but i'm not actually sure where else we use them right now?!

Reply:
definitely use them in comments as well

Reply:
will add something, somewhere. But closing this topic here :)